src/ExternalImage.php

```diff
@@ -0,0 +1,479 @@
+ *
+ * The `Timber\ExternalImage` class represents an external image loaded via http.
+ *
+ * @api
+ * @example
+ * ```php
+ * $context = Timber::context();
+ *
+ * // Lets say you have an external image that you want to use in your theme
+ *
+ * $context['cover_image'] = Timber::get_external_image($url);
+ *
+ * Timber::render('single.twig', $context);
+ * ```
+ *
+ * ```twig
+ * <article>
+ *   <img src="{{cover_image.src}}" class="cover-image" />
+ *   <h1 class="headline">{{post.title}}</h1>
+ *   <div class="body">
+ *     {{post.content}}
+ *   </div>
+ *
+ *  <img
+ *    src="{{ get_image(post.custom_field_with_image_id).src }}"
+ *    alt="Another way to initialize images as Timber\Image objects, but within Twig" />
+ * </article>
+ * ```
+ *
+ * ```html
+ * <article>
+ *   <img src="http://example.org/wp-content/uploads/2015/06/nevermind.jpg" class="cover-image" />
+ *   <h1 class="headline">Now you've done it!</h1>
+ *   <div class="body">
+ *     Whatever whatever
+ *   </div>
+ *   <img
+ *     src="http://example.org/wp-content/uploads/2015/06/kurt.jpg"
+ *     alt="Another way to initialize images as Timber\Image objects, but within Twig" />
+ * </article>
+ * ```
+ */
+class ExternalImage implements ImageInterface
+{
+    /**
+     * File.
+     *
+     * @api
+     * @var string
+     */
+    private $alt_text;
+
+    protected function __construct($alt)
+    {
+        $this->alt_text = $alt;
+    }
+
+    /**
+     * File.
+     *
+     * @api
+     * @var mixed
+     */
+    public $file;
+
+    /**
+     * File location.
+     *
+     * @api
+     * @var string The absolute path to the attachmend file in the filesystem
+     *             (Example: `/var/www/htdocs/wp-content/uploads/2015/08/my-pic.jpg`)
+     */
+    public $file_loc;
+
+    /**
+     * Absolute URL.
+     *
+     * @var string The absolute URL to the attachment.
+     */
+    public $abs_url;
+
+    /**
+     * File extension.
+     *
+     * @api
+     * @since 2.0.0
+     * @var null|string A file extension.
+     */
+    public $file_extension = null;
+
+    /**
+     * Formatted file size.
+     *
+     * @api
+     * @since 2.0.0
+     * @var FileSize File size string.
+     */
+    private $file_size = null;
+
+    /**
+     * File types.
+     *
+     * @var array An array of supported relative file types.
+     */
+    private $image_file_types = [
+        'jpg',
+        'jpeg',
+        'png',
+        'svg',
+        'bmp',
+        'ico',
+        'gif',
+        'tiff',
+        'pdf',
+    ];
+
+    /**
+     * Image dimensions.
+     *
+     * @internal
+     * @var ImageDimensions stores Image Dimensions in a structured way.
+     */
+    protected ImageDimensions $imageDimensions;
+
+    /**
+     * Inits the ExternalImage object.
+     *
+     * @param $url string URL to load the image from.
+     * @param $alt string ALT text for the image.
+     * @internal
+     *
+     */
+    public static function build($url, $alt)
```

Comment:
I would use an options array for the second parameter, that takes alt as array key. Who knows what other options we’ll need in the future?

```php
public static function build($url, array $args = [])
{
    $args = wp_parse_args( $args, [
        'alt' => '',
     ] );

    // …

    $image->set_alt( $args['alt'] );
}
```

Reply:
makes sense. Changed.